src/mruby-zest/example/TextLine.qml

```diff
@@ -50,28 +50,72 @@ Widget {
         (0...l.length).each do |i|
             l[i] = "?" if l.getbyte(i) > 127
         end
-        vg.text(8,h/2,l)
-        bnd = vg.text_bounds(0,0,l)
-        if(@state)
-            vg.text(8+bnd,h/2,"|")
+
+        @edit ||= EditRegion.new($vg, self.label, w-20, h*0.8)
+        @edit.each_string do |x, y, str, cursor|
+            if(cursor == false)
+                vg.text(x+10, y, str)
+            else
+                if(@state)
+                    vg.text_align NVG::ALIGN_LEFT| NVG::ALIGN_MIDDLE
+                    vg.text(x+10, y, str)
+                end
+            end
         end
+
     }
 
     function onKey(k, mode)
     {
         return if mode != "press"
+        pos = self.label.length
+        pos = @edit.pos if @edit
+        ll = self.label
+
+        cursor_row = @edit.calc_cursor_y
```

Comment:
`calc_cursor_y` is not defined in util.rb:EditRegion, which makes the UI crash as soon as you try typing something in the widget.

Have you committed all your changes?

Reply:
Hi @pgervais,
I added another commit to solve this issue and now it should work properly.